### ISSUE TITLE
On Windows, No BGAPI compatible device detected

### ISSUE THREAD

**Issue by designingSparks** [peplin/pygatt#164]
This is new in version 3.2. The program gets stuck at line 163 in bgapi.py due to a serial.serialutil.SerialException. In version 3.0, this problem didn't occur.

I have a temporary fix by commenting out lines 168-170:
`elif attempt == 0:
  raise NotConnectedError(
 "No BGAPI compatible device detected")`

**SOLUTION**
It turns out that this problem can be fixed by adding a short delay of 0.5s before the second call to `self._open_serial_port()` in start() in the file bgapi.py

**Comment by alxnik** [comment]
+1
BGAPIBackend connects to the device and resets it. It then tries to reconnect while the device has not reset yet so it fails. 
Adding a delay is enough to let the device reset.

**Comment by lemarquois** [comment]
+1
same problem here. I also added a delay of 1s

**Comment by draapho** [comment]
see here:
https://github.com/peplin/pygatt/issues/179

change one line in `_open_serial_port` function
``` python
def _open_serial_port(self, max_connection_attempts=MAX_CONNECTION_ATTEMPTS):

        for attempt in range(max_connection_attempts):
            try:
                ................................
            except (BGAPIError, serial.serialutil.SerialException, serial_exception):
                # elif attempt == 0: ######## change this line to 
                elif attempt >= max_connection_attempts:  ##### changed like this 
               ................
```

**Comment by Gra55h0pper** [comment]
@draapho: I think you intended to change it to: 

    elif attempt >= max_connection_attempts-1:  ##### Note the added: '-1'

**Comment by draapho** [comment]
Yes, you are right. Thank you.

**Comment by draapho** [comment]
@Gra55h0pper 
Yes, you are right. Always easy to make a mistake like this. Thank you.

**Comment by peplin** [comment]
This fix landed in v4.0. Thanks for the report and pull request!
